src/EditorFeatures/CSharpTest/ForeachToFor/ForeachToForTests.cs

```diff
@@ -0,0 +1,1177 @@
+    public int Add(object value) => throw new NotImplementedException();
+    public void Clear() => throw new NotImplementedException();
+    public bool Contains(object value) => throw new NotImplementedException();
+    public void CopyTo(Array array, int index) => throw new NotImplementedException();
+    public IEnumerator GetEnumerator() => throw new NotImplementedException();
+    public int IndexOf(object value) => throw new NotImplementedException();
+    public void Insert(int index, object value) => throw new NotImplementedException();
+    public void Remove(object value) => throw new NotImplementedException();
+    public void RemoveAt(int index) => throw new NotImplementedException();
+}
+";
+            await TestInRegularAndScriptAsync(text, expected);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.ForeachToFor)]
+        public async Task ImmutableArray()
+        {
+            var text = @"
+<Workspace>
+    <Project Language=""C#"" AssemblyName=""Assembly1"" CommonReferences=""true"" CommonReferenceFacadeSystemRuntime = ""true"">
+    <MetadataReference>" + typeof(ImmutableArray<>).Assembly.Location + @"</MetadataReference>
+        <Document>
+using System;
+using System.Collections.Immutable;
+
+class Test
+{
+    void Method()
+    {
+        var list = ImmutableArray.Create(1);
+        foreach [||](var a in list)
+        {
+            Console.WriteLine(a);
+        }
+    }
+}</Document>
+    </Project>
+</Workspace>";
+
+            var expected = @"
+using System;
+using System.Collections.Immutable;
+
+class Test
+{
+    void Method()
+    {
+        var list = ImmutableArray.Create(1);
+        for (var {|Rename:i|} = 0; i < list.Length; i++)
+        {
+            var a = list[i];
+            Console.WriteLine(a);
+        }
+    }
+}";
+            await TestInRegularAndScriptAsync(text, expected);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.ForeachToFor)]
+        public async Task ExplicitInterface()
+        {
+            var text = @"
+using System;
+using System.Collections;
+using System.Collections.Generic;
+
+class Test
+{
+    void Method()
+    {
+        var list = new Explicit();
+        foreach [||] (var a in list)
+        {
+            Console.WriteLine(a);
+        }
+    }
+}
+
+class Explicit : IReadOnlyList<int>
+{
+    int IReadOnlyList<int>.this[int index] => throw new NotImplementedException();
+    int IReadOnlyCollection<int>.Count => throw new NotImplementedException();
+    IEnumerator<int> IEnumerable<int>.GetEnumerator() => throw new NotImplementedException();
+    IEnumerator IEnumerable.GetEnumerator() => throw new NotImplementedException();
+}
+";
+
+            var expected = @"
+using System;
+using System.Collections;
+using System.Collections.Generic;
+
+class Test
+{
+    void Method()
+    {
+        var list = new Explicit();
+        var {|Rename:list1|} = (IReadOnlyList<int>)(list);
```

Comment:
parens around '(list)' should not be there.  Incorrect simplification?

Reply:
need to put simplification and nosimpliciation annotation. didn't tweak much there yet.

Reply:
please add that.